src/publisher_service/worker.rs

```diff
@@ -0,0 +1,258 @@
+use {
+    super::{
+        helpers::{
+            get_notification_by_notification_id, get_subscriber_by_subscriber_id, sign_message,
+            update_message_processing_status,
+        },
+        types::{NotificationStates, ProcessingMessageItem, ProjectSigningDetails},
+    },
+    crate::{
+        jsonrpc::{JsonRpcParams, JsonRpcPayload, NotifyPayload},
+        model::helpers::get_project_by_project_id,
+        spec::{NOTIFY_MESSAGE_TAG, NOTIFY_MESSAGE_TTL},
+        types::{Envelope, EnvelopeType0},
+        websocket_service::decode_key,
+    },
+    base64::Engine,
+    relay_rpc::{
+        domain::{ClientId, DecodedClientId, Topic},
+        rpc::Publish,
+    },
+    sqlx::{postgres::PgListener, PgPool, Postgres},
+    std::{
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        },
+        time::Duration,
+    },
+    tracing::{info, instrument, warn},
+};
+
+// TODO: These should be configurable, add to the config
+// Maximum of the parallel messages processing workers
+const MAX_WORKERS: usize = 10;
+// Number of workers to be spawned on the service start to clean the queue
+const START_WORKERS: usize = 10;
```

Comment:
Multiple workers is a good and probably necessary approach to reach e.g. 50 msgs/sec, I didn't think about this.

I was also thinking we still want some way to have a hard limit of 50 msgs/sec which we could tune up as we get comfortable with the relay being able to handle it. However, I suppose my thinking was over-engineered as we could simply start with 1 worker, hope it does less than 50 msgs/sec, and then slowly increase the number of workers.

How many msgs/sec does 1 worker do on your particular machine, and how have you tested this?